### PR TITLE
add {{availableInWorker}} to File System API

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

add {{availableInWorker}} to File System API related pages (special case for `FileSystemSyncAccessHandle` and `FileSystemFileHandle.createSyncAccessHandle()`)

add update `FileSystemDirectoryHandle[@@asyncIterator]()` on removing its link to `FileSystemDirectoryHandle.entries()`

also update group data of File System API by replacing `Window` with `Window`, to using the interface rather than the value

### Additional details

reference to https://developer.mozilla.org/en-US/docs/Web/API/File_System_API and https://fs.spec.whatwg.org/

### CODE REVIEW

**Review by bsmth — COMMENTED** [review]
Just a note: FileSystemSyncAccessHandle looks like it's in dedicated workers only, I don't think we have a way to indicate this (https://fs.spec.whatwg.org/#api-filesystemsyncaccesshandle).

**Review by bsmth — APPROVED** [review]
Looks good to me, thank you! 🎉

**Comment by skyclouds2001** [comment]
> Just a note: FileSystemSyncAccessHandle looks like it's in dedicated workers only, I don't think we have a way to indicate this (https://fs.spec.whatwg.org/#api-filesystemsyncaccesshandle).

Yes, as I issue in https://github.com/mdn/yari/issues/10009, if {{availableInWorker}} could support to specific the type of the worker

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Given the PR for yari is made, let's wait for it and update the parameter of `{{AvailableInWorkers}}` here.

File: files/en-us/web/api/file_system_api/index.md [hunk 1]
```diff
@@ -5,9 +5,12 @@ page-type: web-api-overview
 browser-compat:
   - api.FileSystemHandle
   - api.FileSystemFileHandle
+  - api.FileSystemDirectoryHandle
+  - api.FileSystemWritableFileStream
+  - api.FileSystemSyncAccessHandle
 ---
```
Comment (skyclouds2001) [inline files/en-us/web/api/file_system_api/index.md:11]:
```suggestion
spec-urls: https://fs.spec.whatwg.org/
---
```

**Comment by hamishwillee** [comment]
`{{AvailableInWorkers}}` should be applied at the end of the first section, not at the start like the other headers. This is not documented anywhere but is done consistently that way across most of the reference.

FWIW personally I disagree with this - all headers should be together up the top. But without broad agreement on that we should stick with what has already been done.

**Review by bsmth — APPROVED** [review]
Re-approving in case we don't want to wait for the blocking PR: https://github.com/mdn/yari/pull/10029

**Comment by teoli2003** [comment]
I don't know when Yari's people will return to the holiday period. It would be nice if we didn't need extra work here.

**Comment by skyclouds2001** [comment]
as https://github.com/mdn/yari/pull/10029 has been merged yet, the update to the {{AvailableInWorkers}} macros has done yet

**Review by teoli2003 — APPROVED** [review]
👍  Patience is a virtue. Really happy we can merge this now!

**Pull request merged**